tests/test_templates_embeddings.py

```diff
@@ -215,26 +220,68 @@ def circuit(x=None):
-        """Verifies that pennylane.templates.embeddings.AmplitudeEmbedding() raises an exception if ``wires`` is not
-        iterable."""
+    def test_amplitude_embedding_exception_notnormalized(self):
+        """Verifies that pennylane.templates.embeddings.AmplitudeEmbedding() raises an exception if input is not
+        normalized and 'normalization' is set to False."""
 
         n_subsystems = 5
         dev = qml.device('default.qubit', wires=n_subsystems)
 
         @qml.qnode(dev)
         def circuit(x=None):
-            AmplitudeEmbedding(features=x, wires=3, pad=False, normalize=False)
+            AmplitudeEmbedding(features=x, wires=1, pad=False, normalize=False)
             return qml.expval(qml.PauliZ(0))
 
-        with pytest.raises(ValueError, match='Wires needs to be a list of wires that the embedding uses; got 3.'):
-            circuit(x=[1])
+        with pytest.raises(ValueError, match='AmplitudeEmbedding requires a normalized feature vector. '
+                                             'Set ``normalize=True`` to automatically normalize it.'):
+            circuit(x=[1, 1])
+
+    @pytest.mark.parametrize("inpt, expected", [(np.array([0, 2]), [0, 1]),
+                                                (1/np.sqrt(2)*np.array([2, 2]), [1/np.sqrt(2), 1/np.sqrt(2)]),
+                                                ([np.complex(-np.sqrt(0.1), 0.0), np.sqrt(0.3)], [-0.5, 0.8660254]),
+                                                ([0, 2], [0, 1])])
+    def test_amplitude_embedding_normalizes_keywordargs(self, inpt, expected):
+        """Verifies that pennylane.templates.embeddings.AmplitudeEmbedding() normalizes an input passed as a keyword
+         argument if `normalize` is set to `True`."""
+
+        dev = qml.device('default.qubit', wires=1)
+
+        @qml.qnode(dev)
+        def circuit(x=None):
+            AmplitudeEmbedding(features=x, wires=0, pad=False, normalize=True)
+            return qml.expval(qml.PauliZ(0))
+
+        circuit(x=inpt)
+        state = dev._state
+        assert np.allclose(state, expected)
+
+    @pytest.mark.parametrize("inpt, expected", [(np.array([0, 2]), [0, 1]),
+                                                (1/np.sqrt(2)*np.array([2, 2]), [1/np.sqrt(2), 1/np.sqrt(2)]),
+                                                ([np.complex(-np.sqrt(0.1), 0.0), np.sqrt(0.3)], [-0.5, 0.8660254]),
+                                                ([0, 2], [0, 1])])
+    def test_amplitude_embedding_normalizes_positionalargs(self, inpt, expected):
+        """Verifies that pennylane.templates.embeddings.AmplitudeEmbedding() normalizes an input passed as a positional
+        argument if `normalize` is set to `True`."""
+
+        dev = qml.device('default.qubit', wires=1)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            AmplitudeEmbedding(features=x, wires=0, pad=False, normalize=True)
+            return qml.expval(qml.PauliZ(0))
+
+        circuit(inpt)
```

Comment:
ok I see here that you can pass the input as a positional argument. How come the other tests use the keyword-arg approach?

Reply:
See above.